src/components/P5ChallengeTaskStepper/P5ChallengeTaskStepper.jsx

```diff
@@ -0,0 +1,146 @@
+    navigate("/ChallengePage");
+  };
+  const [currentStep, setCurrentStep] = useState(0);
+
+  const handleNextStep = () => {
+    if (currentStep < 3) {
+      // Assuming you have 3 steps (you can adjust this value based on your step count)
+      setCurrentStep(currentStep + 1);
+    }
+  };
+  return (
+    // <div className="task-stepper-container">
+    //   {/* Connect with Partner Button */}
+    //   <div className="step-button">
+    //     <div className="button-text">Connect with Partner</div>
+    //   </div>
+
+    //   {/* Task Details */}
+    //   <div className="task-details">
+    //     {/* LinkedIn Section */}
+    //     <div className="task-section">
+    //       <div className="section-title">LinkedIn</div>
+    //       <div className="section-description">
+    //         Connect with your partner by using the LinkedIn button located in
+    //         the right tab next to “Connect.”
+    //         <br />
+    //         Alternatively, you can look them up by their username.
+    //       </div>
+    //     </div>
+
+    //     {/* Google Meet Section */}
+    //     <div className="task-section">
+    //       <div className="section-title">Google Meet</div>
+    //       <div className="section-description">
+    //         Use the Google Meet button in the right tab next to “Connect.” This
+    //         will direct you to Google Meet, where a meeting template will be
+    //         prefilled, making it easier to schedule a session with your partner.
+    //       </div>
+    //     </div>
+    //   </div>
+    // </div>
+    <div className="task-stepper-container">
+      {taskDetails.steps.map((step, index) => (
+        <>
+          <div key={index} className="step-section">
+            {/* Dynamic Button for each step */}
+            <div
+              className={`step-button ${currentStep == index ? 'completed' : ''}`}
+              style={{ backgroundColor: step.buttonColor }}
+            >
+              <div className="button-text">{step.stepTitle}</div>
+            </div>
+
+            {/* Task Details */}
+            <div className="task-details">
+              {step.stepContent.map((content, subIndex) => (
+                <div key={subIndex} className="task-section">
+                  <div className="section-title">{content.sectionTitle}</div>
+                  <div
+                    className="section-description"
+                    style={{ whiteSpace: "pre-wrap" }}
+                  >
+                    {content.description}
+                  </div>
+                </div>
+              ))}
+            </div>
+          </div>
+        </>
+      ))}
+      <div className="navigation-bottom-buttons">
+        <button className="dashboard-button" onClick={handleDashboard}>
+          <span className="dashboard-text">Dashboard</span>
+        </button>
+        <button className={`next-button ${currentStep >= 2 ? 'completed' : ''}`} onClick={handleNextStep}>
```

Comment:
These conditional classes aren't currently changing anything. I'd recommend changing to something like this:

`{currentStep >= 3 ?

          <button className={`next-button next-button--completed`}>
            <span className="next-text">Completed</span>
          </button> 
          :
          <button className={`next-button`} onClick={handleNextStep}>
            <span className="next-text">Next Step</span>
          </button>
        }`
        
        This way the button will change both it's color and text when the challenge is complete. 
        
        The .next-button--active would need to be added to the stylesheet as well

Reply:
resolved